lib/unison-sqlite/src/Unison/Sqlite/Connection.hs

```diff
@@ -121,51 +131,96 @@ closeConnection (Connection _ _ conn) =
 
 execute :: Sqlite.ToRow a => Connection -> Sql -> a -> IO ()
 execute conn@(Connection _ _ conn0) s params =
-  Sqlite.execute conn0 (coerce s) params `catch` \(exception :: Sqlite.SQLError) ->
+  doExecute `catch` \(exception :: Sqlite.SQLError) ->
     throwSqliteQueryException
       SqliteQueryExceptionInfo
         { connection = conn,
           exception = SomeSqliteExceptionReason exception,
           params = Just params,
           sql = s
         }
+  where
+    doExecute =
+      if debugTraceQueries
+        then do
+          Text.hPutStrLn stderr ("query:  " <> coerce s)
+          Text.hPutStrLn stderr ("params: " <> Text.pack (anythingToString params))
+          Text.hPutStrLn stderr "----------"
+          run
+        else run
+      where
```

Comment:
Nit, I personally find nested where's like this tougher to read, but if that's your preference that's fine 🤷🏼‍♂️

Reply:
Whoops, it wasn't intentional